Swift SDK migration

ameypv-rudder opened the pull request:
## Description of the change

> Rebasing the older branch was causing some issues so moved the doc in a new PR. > Incorporated all suggestions made by @pallabmaiti as listed in [this] PR.

## Type of documentation

- [x] New documentation
- [ ] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Swift migration doc](https://www.notion.so/rudderstacks/Documentation-Update-Swift-SDK-c7e6e635bbdb4c91970926b20a3c8274)

ameypv-rudder commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+| :---------------| :---------------|
+| `RudderClient` | `RSClient`       |
+| `RudderConfig`| `RSConfig`     |
+| `RudderConfigBuilder` | `RSConfigBuilder` |
```
@pallabmaiti as per your comment [here](https://github.com/rudderlabs/rudderstack-docs/pull/506/files#r874472740), `RSConfigBuilder` class is not available now. Should I go ahead and remove the reference from this table?

pallabmaiti replied:
Yes please.

ameypv-rudder commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+| `RudderConfigBuilder` | `RSConfigBuilder` |
+| `RudderLogLevelDebug` | `RSLogLevelDebug` |
+
+### How do I ensure the events tracked just before closing/backgrounding the app are sent immediately and not on the next app launch?
```
@pallabmaiti as per your comment [here](https://github.com/rudderlabs/rudderstack-docs/pull/506/files#r873504858), we don't support this as of now but we might re-support it later.

Can we remove it from this doc **for now**, and then re-add it once we add support for this feature?

pallabmaiti replied:
Yes, we can do that.

ameypv-rudder commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
````diff
@@ -0,0 +1,1114 @@
+</div>
+
+```objectivec
+RSConfigBuilder *builder = [[RSConfigBuilder alloc] init];
````
@pallabmaiti  Since `RSConfigBuilder` is not supported/available, does this code block need to be changed (if we are retaining this FAQ and not removing it completely)?

pallabmaiti replied:
I think this entire portion will be deleted. So, yes.

ameypv-rudder commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+| [watchOS](https://developer.apple.com/watchos/) | 1.3.0 |
+| [macOS](https://developer.apple.com/macos/) | 2.0.0 |
+
+### How do I migrate from an older SDK version to the current version?
```
@pallabmaiti can you please help me with [this comment](https://github.com/rudderlabs/rudderstack-docs/pull/506/files#r873506200)? How do I fetch the current version?

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+<li><code class="inline-code">Application Opened</code></li>
+<li><code class="inline-code">Application Backgrounded</code></li>
+</ul>
+You can disable these events using the <code class="inline-code">trackLifecycleEvents</code> method of <code class="inline-code">RSConfig</code> by passing <code class="inline-code">false</code>. However, it is highly recommended to keep them enabled.
```
```suggestion
You can disable these events using the <code class="inline-code">trackLifecycleEvents</code> method of <code class="inline-code">RSConfig</code> and passing it as <code class="inline-code">false</code>. However, it is highly recommended to keep them enabled.
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+According to the <a href="https://developer.apple.com/documentation/uikit/uidevice/1620059-identifierforvendor">Apple documentation</a>, if a device has multiple apps from the same vendor, all the apps will be assigned the same <code class="inline-code">deviceId</code>. If all the applications from the vendor are uninstalled, then a new <code class="inline-code">deviceId</code> will be assigned t ...
+</div>
+
+An sample `identify` call is shown below:
```
```suggestion
A sample `identify` call is shown below:
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+
+## Alias
+
+The <Link to="/rudderstack-api/api-specification/rudderstack-spec/alias"><code class="inline-code">alias</code></Link> call associates the user with a new identification. A sample `alias` call for the API is below:
```
```suggestion
The <Link to="/rudderstack-api/api-specification/rudderstack-spec/alias"><code class="inline-code">alias</code></Link> call associates the user with a new identification. A sample `alias` call is shown below:
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+    </TabPanels>
+</Tabs>
+
+Alternatively, you can use the following method signature
```
```suggestion
Alternatively, you can use the following method signature:
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+
+## Reset
+
+You can use the `reset` method to clear the persisted user traits for the `identify` call. This is required for the user logout operation.
```
```suggestion
You can use the `reset` method to clear the persisted user traits from the `identify` call. This is required for the user logout operation.
```

ameypv-rudder commented:
@pallabmaiti 

- Changed “Swift” to “iOS v2" and restructured the folder accordingly.
- Added information on the new features in the main iOS doc.
- Added relevant versions in the new and the current doc.

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+
+<div class="infoBlock">
+
+You only need to call the <code class="inline-code">setoptOutStatus</code> API with the relevant parameter only once, as the information persists within the device even if you reboot it.
```
```suggestion
You need to call the <code class="inline-code">setoptOutStatus</code> API with the relevant parameter only once, as the information persists within the device even if you reboot it.
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+You only need to call the <code class="inline-code">setoptOutStatus</code> API with the relevant parameter only once, as the information persists within the device even if you reboot it.
+</div>
+
+The following snippet highlights the use of the `optOut` API to disable user tracking:
```
```suggestion
The following snippet highlights the use of the `setoptOutStatus` API to disable user tracking:
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-swift-sdk.mdx:
```diff
@@ -0,0 +1,1114 @@
+    </TabPanels>
+</Tabs>
+
+Once the user grants their consent, you can enable user tracking once again by using the `setOptOutStatus` API with `NO` or `false` as a parameter sent to it, as shown:
```
```suggestion
Once the user grants their consent, you can enable user tracking once again by using the `setOptOutStatus` API with `NO` or `false` as its parameter, as shown:
```

pallabmaiti commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:
```diff
@@ -0,0 +1,1101 @@
+[option putIntegration:@"Amplitude" isEnabled:YES];
+[option putIntegration:@"<DESTINATION_DISPLAY_NAME>" isEnabled:<BOOLEAN>];
+// specifying destination by its Factory instance
+[option putIntegrationWithFactory:[RudderMoengageFactory instance] isEnabled:NO];
```
Line no. 756 & 757 is not available anymore.

pallabmaiti commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:
```diff
@@ -0,0 +1,1101 @@
+option.putIntegration("Amplitude", isEnabled:true)
+option.putIntegration(<DESTINATION_DISPLAY_NAME>, isEnabled:<BOOLEAN>)
+// specifying destination by its Factory instance
+option.putIntegration(with: RudderMoengageFactory.instance(), isEnabled: true);
```
Line no. 772 & 773 not available anymore.

pallabmaiti commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2.mdx:
````diff
@@ -0,0 +1,1101 @@
+2. Then, create a `CustomDetination` class file by extending `RudderDestination` and initialize `RSCustomDestination` inside `init()`, as shown:
+
+```swift
+class CustomDestination: RudderDestination {
````
```suggestion
@objc
class CustomDestination: RudderDestination {
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/index.mdx:
```diff
@@ -13,14 +13,27 @@ The **RudderStack iOS SDK** lets you track your customer event data from your iO
-<div class="successBlock">
+<div class="infoBlock">
+
+<strong>This documentation is written for the iOS SDK v1.6.0 and below</strong>. <br /><br />
```
```suggestion
<strong>This document is written for the iOS SDK v1.6.0 and below</strong>. <br /> 
You can also integrate the iOS SDK with your tvOS and watchOS apps and seamlessly track user events without any additional configuration. 

The tvOS apps are supported in the SDK version <strong>1.1.0 and above</strong> whereas the watchOS apps are supported in the SDK version <strong>1.3.1 and above</strong>.

</div>
```

ameypv-rudder replied:
```suggestion
<strong>This documentation is written for the iOS SDK v1.x</strong>. <br /><br />
```

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/index.mdx:
```diff
@@ -13,14 +13,27 @@ The **RudderStack iOS SDK** lets you track your customer event data from your iO
+<div class="infoBlock">
+
+<strong>This documentation is written for the iOS SDK v1.6.0 and below</strong>. <br /><br />
+
```
<div class="infoBlock">

ishachopra0401 commented on docs/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/index.mdx:
```diff
@@ -13,14 +13,27 @@ The **RudderStack iOS SDK** lets you track your customer event data from your iO
+For more information, refer to the <Link to="/stream-sources/rudderstack-sdk-integration-guides/rudderstack-ios-sdk/ios-v2">iOS SDK v2</Link> documentation.
+</div>
+
+<div class="infoBlock">
```
This info block can be removed as content is merged above.

The idea is to keep the content for a specific iOS SDK version together.
